src/main/java/com/example/java/maven/game2048/Board.java

```diff
@@ -18,18 +23,267 @@ public Board(int numberOfRows, int numberOfColumns) {
+                rightOrDownSort(column);
+            }
+        }
+    }
+
+    public void sumLeft() {
+        for (Field[] row : this.fields) {
+            sumInRowOrColumnLeftOrUp(row);
+        }
+    }
+
+    public void sumDown() {
+        for (Field[] column : getColumns()) {
+            sumInRowOrColumnRightOrDown(column);
+        }
+    }
+
+    public void sumUp() {
+        for (Field[] column : getColumns()) {
+            sumInRowOrColumnLeftOrUp(column);
+        }
+    }
+
+    public void sumRight() {
+        for (Field[] row : this.fields) {
+            sumInRowOrColumnRightOrDown(row);
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(numberOfRows, numberOfColumns);
+        result = 31 * result + Arrays.hashCode(fields);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Board board = (Board) o;
+        return numberOfRows == board.numberOfRows &&
+                numberOfColumns == board.numberOfColumns &&
+                Arrays.deepEquals(fields, board.fields);
+    }
+
     @Override
     public String toString() {
         StringBuilder strBoardBuilder = new StringBuilder();
         for (Field[] row : this.fields) {
             strBoardBuilder.append(
                     Arrays.toString(row)
-                            .replace(',', ' ')
-                            .replace('[', ' ')
-                            .replace(']', ' ')
+                            .replace(",", "")
+                            .replace("[", "")
+                            .replace("]", "")
             );
             strBoardBuilder.append("\n");
         }
-        return strBoardBuilder.toString();
+        return strBoardBuilder.toString().trim();
+    }
+
+    public boolean checkIfWon(int winningNumber) {
+        for (Field[] row : this.fields) {
+            for (Field field : row) {
+                if (field.getValue() == winningNumber) {
+                    System.out.print(this);
+                    System.out.println("YOU WON!");
+                    return true;
+
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean checkIfSimilarNeighboringFields() {
+        for (Field[] row : this.fields) {
+            for (int index = 0; index < row.length - 1; index++) {
+                if (row[index].getValue() == row[index + 1].getValue()) {
+                    return true;
+                }
+            }
+        }
+        for (Field[] column : getColumns()) {
+            for (int index = 0; index < column.length - 1; index++) {
+                if (column[index].getValue() == column[index + 1].getValue()) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean checkIfGameOver() {
```

Comment:
as mentioned in earlier comment - this method should NOT be in board. You probably need to add methods to board like:
- `int getNumberOfFieldsWithZeroValues()`
- `boolean hasSimilarNeighboringFields()`
(NOT SURE :D think about it that this methods are necessary :D )

Reply:
Done